staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1691,6 +1691,8 @@ message WindowsMemoryUsage {
     UInt64Value available_bytes = 3;
```

Comment:
is available_bytes definition still `limit - workingSetBytes`? Would it be `limit - committed_memory`?

Reply:
good question, @kiashok do you know?

Reply:
what is "limit" here?

Reply:
can we change it as part of this PR?

Reply:
> @marosset can you please comment on [#117127 (comment)](https://github.com/kubernetes/kubernetes/pull/117127#discussion_r1212

Reply:
Change the comment in here?

Reply:
Yes, I do think we will want to update **available_bytes** to be `limit - committed_memory` once commit memory is plumbed through all the layers.

Windows containers enforce use JOBOBJECTS to enforce memory usage and the job objects use commit memory for enforcement.

From https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-jobobject_basic_limit_information

JOB_OBJECT_LIMIT_JOB_MEMORY
> Causes all processes associated with the job to limit the job-wide sum of their committed memory. When a process attempts to commit memory that would exceed the job-wide limit, it fails. If the job object is associated with a completion port, a JOB_OBJECT_MSG_JOB_MEMORY_LIMIT message is sent to the completion port.
This limit requires use of a [JOBOBJECT_EXTENDED_LIMIT_INFORMATION](https://learn.microsoft.com/en-us/windows/desktop/api/winnt/ns-winnt-jobobject_extended_limit_information) structure. Its BasicLimitInformation member is a JOBOBJECT_BASIC_LIMIT_INFORMATION structure.

JOB_OBJECT_EXTENDED_LIMIT_INFORMATION also acts on commited memory
https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-jobobject_extended_limit_information

> If the LimitFlags member of the [JOBOBJECT_BASIC_LIMIT_INFORMATION](https://learn.microsoft.com/en-us/windows/desktop/api/winnt/ns-winnt-jobobject_basic_limit_information) structure specifies the JOB_OBJECT_LIMIT_JOB_MEMORY value, this member specifies the limit for the virtual memory that can be committed for the job. Otherwise, this member is ignored.

We'll need to figure out the best way to safely change this behavior but the first step is updating the CRI so can start having container runtimes report this information.

Reply:
are these stats in alpha? Maybe a good idea to change to the desired state and comment about the old version behavior.

Reply:
I think pre-alpha (at least for K8s)?
We need to update the CRI before we can expose the stats in containerd.

Reply:
if you plan to change what the `available_bytes ` will report, I'd suggest to change the comment now (and maybe have a small comment saying that oder version may return a different value). Given the status of it - should be ok to change it

Reply:
so what is the plan? Keep the comment as it even though it is not working like this today and the plan to change the implementation? I am a bit lost in what the conclusion here is.

Reply:
The idea is to expose commit memory through CRI and populate available_bytes first and then reflect the changes by updating the code comment here. It would make sense to update the comment after the stats are actually populated and returned to avoid any confusions too.

Reply:
I don't follow. Do you want to populate `available_bytes` according to the existing comment or with what it should be? If first - why not just change the definition now and populate it with the new definition?

Reply:
We will change it to be what it should (commit memory) and after that update the comment.

Reply:
@SergeyKanzhelev could you please take a look at this PR when you have time? Thanks!